egui-wgpu/src/renderer.rs

```diff
@@ -384,6 +378,9 @@ impl RenderPass {
             match primitive {
                 Primitive::Mesh(mesh) => {
                     if let Some((_texture, bind_group)) = self.textures.get(&mesh.texture_id) {
+                        let index_buffer = index_buffers.next().unwrap();
+                        let vertex_buffer = vertex_buffers.next().unwrap();
```

Comment:
In what circumstance would this `unwrap` crash?

Reply:
It can only crash if we had a bug in `update_buffers()` ( [here](https://github.com/emilk/egui/pull/1704/files/6160a9507f3c4afe18a8b49cc4cb80a1d00bd3aa#diff-9a3a7bbf3e24ed9aa455942f77365cfe970b486b43f9f1d97d00ca56ab7cfba2R669) ). If we failed to push a vertex and index buffer to the list for every `Primitive::Mesh`, then the unwrap would panic.

It can't fail because of any action on the part of the user of egui.

Before this PR, if there was a bug in `update_buffers()` it would have just silently not rendered certain meshes, because `zip()`-ing iterators of different sizes will just stop at the shortest iterator. Now it is actually better, because if we mess that up, it will crash, instead of silently and randomly fail to render certain meshes. That's actually what made this bug difficult to find.